odoo/tools/xml_utils.py

```diff
@@ -83,3 +83,74 @@ def create_xml_node(parent_node, node_name, node_value=None):
     :returns (etree._Element):
     """
     return create_xml_node_chain(parent_node, [node_name], node_value)[0]
+
+
+def cleanup_xml_node(xml_node_or_string, remove_comments=True, remove_blank_text=True, remove_blank_nodes=True, indent_level=0, indent_space="  "):
+    """Clean up the sub-tree of the provided XML node.
+
+    If the provided XML node is of type:
+    - etree._Element, it is modified in-place.
+    - string/bytes, it is first parsed into an etree._Element
+    :param xml_node_or_string (etree._Element, str): XML node (or its string/bytes representation)
+    :param remove_comments (bool): if True, removes comment nodes
+    :param remove_blank_text (bool): if True, removes whitespace-only text from nodes
+    :param remove_blank_nodes (bool): if True, removes leaf nodes with no text (iterative, depth-first, done after remove_blank_text)
+    :param indent_level (int): depth or level of node within root tree (use -1 to leave indentation as-is)
+    :param indent_space (str): string to use for indentation (use '' to remove all indentation)
+    :returns (etree._Element): clean node, same instance that was received (if applicable)
+    """
+    xml_node = xml_node_or_string
+
+    # Input is str/bytes: parser takes care of removing blanks/comments
+    if isinstance(xml_node, (str, bytes)):
+        parser = etree.XMLParser(remove_blank_text=remove_blank_text, remove_comments=remove_comments)
+        if isinstance(xml_node, str):
+            xml_node = xml_node.encode()  # misnomer: fromstring actually reads bytes
+        xml_node = etree.fromstring(xml_node, parser=parser)
+        remove_comments = False  # indicates job is done
+        # https://lxml.de/FAQ.html: remove_blank_text only removes some blank texts
+
+    # Process leaf nodes iteratively
+    # Depth-first, so any inner node may become a leaf too (if children are removed)
+    def leaf_iter(parent_node, node):
+        for child_node in node:
+            leaf_iter(node, child_node)
+
+        # Base condition: node is not root nor inner node
+        if parent_node is not None and not len(node):
+            if remove_blank_text and not (node.text or '').strip():
+                if node.text is not None:  # keep self-closing tags that way
```

Comment:
Actually, the only value other than string that node.text can have is None, so the two conditions could be replaced by:
`remove_blank_text and node.text is not None and not node.text.strip()`